### PR TITLE
Event Hubs message broker migration support

### PR DESCRIPTION
**Pull request by andrewstanovsky**
## Description

Support for Kafka to Event Hubs migration and future EH to EH migrations in FRS.
The change itself is transparent for Kafka and will be changing behavior only in case of Event Hubs activation

## Reviewer Guidance

I had to move messageBrokerId assignment out of ISession object initialization to avoid explicitly assigning "undefined" value right within an object. If you do it, then mongoDB will serialize "undefined" as "null" - which would break the logic for EH migration. It's also possible to provide 'ignoreUndefined: true' option for the mongodb client that fixes the problem, but it could break existing code that might rely on "undefined" being serialized as "null"

Verified session transition scenarios:

- Same FRS cluster: Kafka to Event Hubs, Event Hubs to Kafka, Event Hubs to Event Hubs (different instance)
- Between FRS clusters of the same group: Kafka to Kafka, Kafka to Event Hubs, Event Hubs to Kafka

### CODE REVIEW

File: server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts [hunk 1]
```diff
@@ -227,6 +235,8 @@ export function create(
 						isSessionAlive: false,
 						isSessionActive: false,
 					};
+					// if undefined and added directly to the session object - will be serialized as null in mongo which is undesirable
```
Comment (manishgargd) [inline server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts:238]: nit: where is the session object serialized in Mongo?

Comment (andrewstanovsky) [reply server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts:238]: In this case it's not that important and it can be inlined. But I decided to assign it outside in case someone's refactor this code and decides to add db write of this object. So just to be on a safe side :)

**Review by pradeepvairamani — APPROVED** [review]

File: packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md [hunk 2]
```diff
@@ -4,8 +4,6 @@
 
 ```ts
 
-/// <reference types="mocha" />
```
Comment (pradeepvairamani) [inline packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md:7]: Why was this change needed? Just want to make sure that this isn't inadvertent

Comment (andrewstanovsky) [reply packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md:7]: That's what prettier decided to change for some reason. I'm not sure how it ended up merged in if it has to pass prettier. Maybe something has changed in its config

File: server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts [hunk 3]
```diff
@@ -227,6 +235,8 @@ export function create(
...
+					if (messageBrokerId) session.messageBrokerId = messageBrokerId;
```
[the remark this answers is not in the document]
Comment (andrewstanovsky) [reply server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts:239]: Ok, will fix that :)

[the remark this answers is not in the document]
Comment (andrewstanovsky) [reply server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts]: Fixed

File: server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts [hunk 4]
```diff
@@ -150,6 +160,8 @@ async function updateExistingSession(
 		// If session was not alive, it cannot be "active"
 		isSessionActive: false,
 	};
+	// if undefined and added directly to the session object - will be serialized as null in mongo which is undesirable
+	if (updatedMessageBrokerId) updatedSession.messageBrokerId = updatedMessageBrokerId;
```
Comment (pradeepvairamani) [inline server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts:164]: One more place :)

Comment (andrewstanovsky) [reply server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts:164]: Oops, missed that one :) on it

**Review by manishgargd — APPROVED** [review]
